### ISSUE TITLE
Aircraft related rates problems

### ISSUE THREAD

**Issue by sean-fitzpatrick** [APEXCalculus/APEXCalculusPTX#228]
This is mainly a question for @APEXCalculus but perhaps others also have opinions.

I'm updating the sections on related rates and optimization because we can now include TikZ images in the WeBWorK problems.
There are a couple of questions about aircraft flying past anti-aircraft guns.

Right now, some numbers are given in feet, and some in miles. Would it be acceptable if we changed things so that everything is in feet? These days I think very few people outside the USA (and maybe Liberia?) are familiar with how many feet are in a mile. I had to look it up just now.

I think the version that Geoff Cox wrote for WeBWorK does everything in feet. My experience with the WeBWorK vesion is that students have enough trouble with the trig that throwing in unit conversions (unfamiliar unit conversions, in most of the world) over-complicates things.

Do we want to keep the miles-to-feet conversion? (If yes, can we add a note that there are 5280 feet in a mile?)

Note that I am resisting the temptation to suggest that we just make these metric... Oops. I guess I just did.

**Comment by sean-fitzpatrick** [comment]
Actually, there's a question for @Alex-Jordan as well, about the WeBWorK code.

The problem with the image has much older WW code (commented out) that looks like:
```
Context()->flags->set(reduceConstants=>0);
$a = Compute("10000/5280*500/1^2*cos(atan(10000/5280/1))^2");
$a = NumberWithUnits("$a rad/hr");
$b = Compute("10000/5280*500/0.2^2*cos(atan(10000/5280/0.2))^2");
$b = NumberWithUnits("$b rad/hr");
$c = Compute("500/(10000/5280)"); <!--Simplify the formula and get this as the limit>-->
$c = NumberWithUnits("$c rad/hr");
```

I added randomization to get
```
Context()->flags->set(reduceConstants=>0);
$spd = random(300,800,10);
$elev = random(5000,15000,100);
$a = Compute("($elev*$spd*5280)/(5280^2+$elev^2)");
$a = NumberWithUnits("$a rad/hr");
$b = Compute("($elev*$spd*5280)/(1056^2+$elev^2)");
$b = NumberWithUnits("$b rad/hr");
$c = Compute("$spd*5280/$elev"); <!--Simplify the formula and get this as the limit>-->
$c = NumberWithUnits("$c rad/hr");
```
(Note that I also updated the previous formulas for the answer, which were overly complicated.)

But the next question, which didn't have an image, has more recently updated WW code:
```
Context()->flags->set(reduceConstants=>0);
$speed = random(400,600,50);
$speedU = NumberWithUnits("$speed mi/h");
@a = (random(1000,2000,100), random(100,500,50), 0);
if($envir{problemSeed}==1){$a[0]=1000,$a[1]=100;};
@aU = map{NumberWithUnits("$_ ft")}(@a);
@b = map{100*$speed*5280/3600/(100**2 + $_**2)}(@a);
@bU = map{NumberWithUnits("$_ rad/s")}(@b);
```

Mostly I'm not familiar with using `@a` for a variable name rather than `$a`. I think this is probably Perl syntax that I don't understand. Can you clear up the difference?

**Comment by sean-fitzpatrick** [comment]
I figured out Alex's fancier code. The `@` creates an array, so this saves a few lines of writing.

**Comment by Alex-Jordan** [comment]
Right, perl has:

$a is a scalar
@a is an array
%a is a hash (like a python dictionary)

@a = (random(1000,2000,100), random(100,500,50), 0);

makes an array with three things. They can be accessed as scalars this way:
$a[0]
$a[1]
$a[2]

(Note that $a is unrelated to @a and $a[0], which can be a source of
confusion.)

@aU = map{NumberWithUnits("$_ ***@***.***);

Applies a map to the elements of @a, where $_ is the dummy variable as you
iterate through the array. The result is stored in a new array @aU.




On Thu, May 25, 2023 at 11:42 AM Sean Fitzpatrick ***@***.***>
wrote:

> I figured out Alex's fancier code. The @ creates an array, so this saves
> a few lines of writing.
>
> —
> Reply to this email directly, view it on GitHub
> <https://protect2.fireeye.com/v1/url?k=31323334-501cfaeb-3132feb7-454455535732-d834720f932d290e&q=1&e=a544dfae-cb67-4232-a4f1-8096e642a98e&u=https%3A%2F%2Fgithub.com%2FAPEXCalculus%2FAPEXCalculusPTX%2Fissues%2F228%23issuecomment-1563340930>,
> or unsubscribe
> <https://protect2.fireeye.com/v1/url?k=31323334-501cfaeb-3132feb7-454455535732-bed106a8b15efae0&q=1&e=a544dfae-cb67-4232-a4f1-8096e642a98e&u=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FABEDOAETBBFYUXPT64RSUHDXH6RY5ANCNFSM6AAAAAAYN57MZ4>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Alex Jordan
Mathematics Instructor
Portland Community College

**Comment by sean-fitzpatrick** [comment]
I'm going to close this, since I think it's resolved by #229. But I should note that some of the problems are not randomized. We may want to revisit that at some point.
